lighthouse-cli/test/smokehouse/readme.md

```diff
@@ -0,0 +1,76 @@
+# Smokehouse
+
+Smokehouse is the Lighthouse end-to-end/smoke test runner. It takes in a set of URLs (usually of custom-built test sites) and runs Lighthouse on them, then compares the results against an expectations file.
+
+By default this is done using the Lighthouse CLI (to exercise the full pipeline) with the tests listed in `smokehouse/test-definitions/core-tests.js`.
+
+## Options
+
+See [`SmokehouseOptions`](https://github.com/GoogleChrome/lighthouse/blob/master/lighthouse-cli/test/smokehouse/smokehouse.js#L23).
+
+## Pipeline
+
+Frontends                                                   Runners
++------------+
+|            |
+|  bin (CLI) +----+                                +--------------+
+|            |    |                                |              |
++------------+    |                            +-->+lighthouse-cli|
+                  |                            |   |              |
++------------+    |      +---------------+     |   +--------------+
+|            |    |      |               |     |
+|   node     +----+----->+ smokehouse.js +-----+
+|            |    |      |               |     |   +--------------+
++------------+    |      +---------------+     |   |              |
+                  |                            +-->+  bundled-LH  |
++------------+    |                                |              |
+|            |    |                                +--------------+
+|bundle-entry+----+
+|            |
++------------+
+
+### Frontend
+- `smokehouse-bin.js` - runs smokehouse from the command line
+- `smokehouse.js` - smokehouse itself is runnable from node
+- TODO: bundle-entry - simple entrypoint to smokehouse that serves as a bundle entry point to run it in a browser.
+
+### Smokehouse
+- `smokehouse.js` - takes a set of smoke-test definitions and runs them via a passed-in runner. If its inputs are bundleable then smokehouse is as well.
+
```

Comment:
```
### Firehouse

Run the core smoke tests, allowing for modifications of expectations to match environment quirks.
```

are you imagining a separate apparatus (firehouse) for this behavior (+ filtering, skipping), or do you plan on moving these options to the smokehouse.js runner?

Reply:
> are you imagining a separate apparatus (firehouse) for this behavior (+ filtering, skipping), or do you plan on moving these options to the smokehouse.js runner?

I'm currently imagining filtering/skipping in the different frontend files, parallel to how the command line version does filtering on test id